app/validators/routing/new_routing_validator.py

```diff
@@ -0,0 +1,77 @@
+from app.validators.routing.new_when_rule_validator import NewWhenRuleValidator
+from app.validators.validator import Validator
+
+
+class NewRoutingValidator(Validator):
+    ROUTING_RULES_DO_NOT_HAVE_A_DEFAULT_RULE = (
+        "Routing rules do not have a default rule"
+    )
+    ROUTING_RULES_HAS_TOO_MANY_DEFAULTS = "Routing rules has more than one default rule"
+    ROUTING_RULE_BLOCK_TARGET_INVALID = "Routing rule routes to invalid block"
+    ROUTING_RULE_GROUP_TARGET_INVALID = "Routing rule routes to invalid group"
+
+    def __init__(self, routing_rules, group, origin_id, questionnaire_schema):
+        super().__init__(routing_rules)
+        self.routing_rules = routing_rules
+        self.group = group
+        self.origin_id = origin_id
+        self.questionnaire_schema = questionnaire_schema
+
+    def validate(self):
+        self.validate_routing_rules_have_default(self.routing_rules)
+
+        block_ids = [block["id"] for block in self.group["blocks"]]
+
+        for rule in self.routing_rules:
+            if "block" in rule:
+                self.validate_routing_rule_block_target(rule, block_ids)
+            elif "group" in rule:
+                self.validate_routing_rule_group_target(
+                    rule, self.questionnaire_schema.group_ids
+                )
+            self.validate_routing_rule(rule)
+
+        return self.errors
+
+    def validate_routing_rules_have_default(self, rules):
+        """
+        Ensure that a set of routing rules contains a default, without a when clause.
+        """
+        default_routing_rule_count = 0
```

Comment:
Could use list comprehension here?
```suggestion
        default_routing_rule_count = sum("when" not in rule for rule in rules)
```

Reply:
Neat! I didn't realise `sum` could be used to count booleans.